src/en/spyfakku/src/eu/kanade/tachiyomi/extension/en/spyfakku/SpyFakku.kt

```diff
@@ -45,28 +53,77 @@ class SpyFakku : HttpSource() {
         .set("Origin", baseUrl)
 
     override fun popularMangaRequest(page: Int): Request {
-        return GET("$baseApiUrl/library?sort=released_at&page=$page", headers)
+        return GET("$baseUrl$baseApiEnd?sort=released_at&page=$page", headers)
     }
 
     override fun popularMangaParse(response: Response): MangasPage {
         val library = response.parseAs<HentaiLib>()
+        val data = getMangasDatas(library.data)
 
-        val mangas = library.archives.map(::popularManga)
+        val mangas = data.first.map(::popularManga)
 
-        val hasNextPage = library.archives.isNotEmpty()
+        val hasNextPage = data.second
 
         return MangasPage(mangas, hasNextPage)
     }
 
+    private fun getMangasDatas(data: List<JsonElement>): Pair<List<ShortHentai>, Boolean> {
```

Comment:
you could make a custom serializer but that can be a bit complicated

Reply:
Hello, Thanks for the suggestion! 
Sorry, but could you please explain more on how I to do that?

Reply:
https://github.com/Kotlin/kotlinx.serialization/blob/master/docs/serializers.md#custom-serializers

tho it is fine without it too

Reply:
Hello, thanks for linking the explanation! 
Sorry, but I think it's too complex for me. 🙏